src/core/atomic.d

```diff
@@ -190,6 +254,14 @@ else version( AsmX86_32 )
         //
         // +=   -=  *=  /=  %=  ^^= &=
         // |=   ^=  <<= >>= >>>=    ~=
+        static if( op == "+=" && __traits(isIntegral, T) && is(T==V1) ) {
```

Comment:
size_t tmp = mod; // convert all operands to size_t
^ Does this incur any performance hit to copy into tmp? Would the below work and possibly be faster:
atomicFetchAdd(T)( ref shared T val, **size_t** mod )